src/benchq/resource_estimation/graph/worstcase_footprint_estimator.py

```diff
@@ -53,10 +54,10 @@ def estimate(
         # simulate graph data
         simulated_graph_data = GraphData(
             max_graph_degree=estimated_max_graph_degree,
-            n_nodes=algorithm_implementation.program.n_nodes,
+            n_nodes=algorithm_implementation.program.min_n_nodes,
```

Comment:
Is `min_n_nodes` a lower bound for `n_nodes`? For the worst case, would we actually want an upper bound on `n_nodes`?

Reply:
Yup! There is a lower bound on n_nodes. But it doesn't really factor into the worst case footprint analysis because we were just assuming that the compilation is done efficiently. The "worst case" part more has to do with the ciruit itself having the worst possible structure, not having a bad compiler that gives an inefficient compilation of the circuit.